apps-rendering/src/client/newsletterSignupForm.ts

```diff
@@ -103,11 +118,29 @@ function setup(form: Element): void {
 	});
 }
 
-function initSignupForms(): void {
-	const signupForms = Array.from(
-		document.querySelectorAll(`form.${COMPONENT_BASE_CLASSNAME}`),
-	);
-	signupForms.forEach(setup);
+function removeContainer(container: Element): void {
+	container.parentElement?.removeChild(container);
+}
+
+function revealContainer(container: HTMLElement): void {
+	container.style.display = 'block';
+}
+
+async function initSignupForms(): Promise<void> {
+	const version = await getBridgetVersion();
+	const signupContainers = Array.from(
+		document.querySelectorAll(`.${COMPONENT_CONTAINER_CLASSNAME}`),
+	) as HTMLElement[];
```

Comment:
In order to avoid the type cast here you could use a type predicate [^1]. TypeScript is able to figure out and narrow the type of `signupContainers` to `HTMLElement[]` via the `filter`.

```suggestion

const isHTMLElement = (element: Element): element is HTMLElement =>
	element instanceof HTMLElement;

async function initSignupForms(): Promise<void> {
	const version = await getBridgetVersion();
	const signupContainers = Array.from(
		document.querySelectorAll(`.${COMPONENT_CONTAINER_CLASSNAME}`),
	).filter(isHTMLElement);
```

[^1]: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates